--- zsh-4.2.3/Completion/Redhat/Command/_rpm	2005-01-12 10:55:18.000000000 +0000
+++ zsh-4.2.3.new/Completion/Redhat/Command/_rpm	2005-01-16 09:18:15.183295304 +0000
@@ -100,6 +100,7 @@
     '(-b)-t+[build mode (tar file)]:build stage:((p\:execute\ \%prep\ stage l\:do\ a\ list\ check c\:execute\ build\ stage i\:execute\ install\ stage b\:build\ a\ binary\ package a\:build\ binary\ and\ source\ packages)):*:build:->build_t' \
     '--rmsource:*:spec file:->spec_files' \
     --{rebuild,recompile}':*:source rpm file:->package_src' \
+    '--define:macro:->macros' \
     '--eval:macro:->macros' && ret=0
 
   # As long as we have a state name...
