diff -Nru zsh-3.1.7.orig/Src/Makefile.in zsh-3.1.7/Src/Makefile.in
--- zsh-3.1.7.orig/Src/Makefile.in	Wed May 31 13:55:56 2000
+++ zsh-3.1.7/Src/Makefile.in	Wed May 31 13:58:05 2000
@@ -178,14 +178,7 @@
 # install binary, creating install directory if necessary
 install.bin-here: zsh$(EXEEXT) install.bin-@L@
 	$(sdir_top)/mkinstalldirs $(DESTDIR)$(bindir)
-	$(INSTALL_PROGRAM) $(STRIPFLAGS) zsh$(EXEEXT) $(DESTDIR)$(bindir)/$(tzsh)-$(VERSION)$(EXEEXT)
-	if test -f $(DESTDIR)$(bindir)/$(tzsh)$(EXEEXT); then \
-	    rm -f $(DESTDIR)$(bindir)/$(tzsh).old; \
-	    $(LN) $(DESTDIR)$(bindir)/$(tzsh)$(EXEEXT) $(DESTDIR)$(bindir)/$(tzsh).old; \
-	else :; fi
-	rm -f $(DESTDIR)$(bindir)/$(tzsh).new
-	$(LN) $(DESTDIR)$(bindir)/$(tzsh)-$(VERSION)$(EXEEXT) $(DESTDIR)$(bindir)/$(tzsh).new
-	mv $(DESTDIR)$(bindir)/$(tzsh).new $(DESTDIR)$(bindir)/$(tzsh)$(EXEEXT)
+	$(INSTALL_PROGRAM) $(STRIPFLAGS) zsh $(DESTDIR)$(bindir)
 .PHONY: install.bin-here
 
 install.bin-N:
