diff -Nru zsh-3.1.6-dev-22.orig/Src/Makefile.in zsh-3.1.6-dev-22/Src/Makefile.in
--- zsh-3.1.6-dev-22.orig/Src/Makefile.in	Thu Apr 13 20:41:06 2000
+++ zsh-3.1.6-dev-22/Src/Makefile.in	Thu Apr 13 20:41:41 2000
@@ -178,14 +178,7 @@
 # install binary, creating install directory if necessary
 install.bin-here: zsh install.bin-@L@
 	$(sdir_top)/mkinstalldirs $(DESTDIR)$(bindir)
-	$(INSTALL_PROGRAM) $(STRIPFLAGS) zsh $(DESTDIR)$(bindir)/$(tzsh)-$(VERSION)
-	if test -f $(DESTDIR)$(bindir)/$(tzsh); then \
-	    rm -f $(DESTDIR)$(bindir)/$(tzsh).old; \
-	    $(LN) $(DESTDIR)$(bindir)/$(tzsh) $(DESTDIR)$(bindir)/$(tzsh).old; \
-	else :; fi
-	rm -f $(DESTDIR)$(bindir)/$(tzsh).new
-	$(LN) $(DESTDIR)$(bindir)/$(tzsh)-$(VERSION) $(DESTDIR)$(bindir)/$(tzsh).new
-	mv $(DESTDIR)$(bindir)/$(tzsh).new $(DESTDIR)$(bindir)/$(tzsh)
+	$(INSTALL_PROGRAM) $(STRIPFLAGS) zsh $(DESTDIR)$(bindir)
 .PHONY: install.bin-here
 
 install.bin-N:
