diff -Nru zsh-3.1.7.orig/configure.in zsh-3.1.7/configure.in
--- zsh-3.1.7.orig/configure.in	Wed May 31 14:57:54 2000
+++ zsh-3.1.7/configure.in	Wed May 31 15:01:33 2000
@@ -515,8 +515,8 @@
 dnl Prefer BSD termcap library to SysV curses library, except on certain
 dnl versions of AIX and HP-UX.
 case "$host_os" in
-  aix*|hpux10.*|hpux11.*) termcap_curses_order="curses ncurses termcap" ;;
-  *)             termcap_curses_order="termcap curses ncurses" ;;
+  aix*|hpux10.*|hpux11.*) termcap_curses_order="curses ncurses termcap tinfo" ;;
+  *)             termcap_curses_order="termcap curses ncurses tinfo" ;;
 esac
 
 for lib in $termcap_curses_order; do
